react/InfosCarrousel/index.jsx

```diff
@@ -18,6 +18,17 @@ const InfosCarrousel = ({ children, theme, className, swipeableProps }) => {
   const hasPreviousInfos = index === 0
   const hasNextInfos = index === count - 1
 
+  const onChangeIndex = useMemo(
+    () =>
+      function(index) {
+        const { onChangeIndex: onChangeIndexProp } = swipeableProps
+        if (onChangeIndexProp) {
+          onChangeIndexProp(index)
+        }
+        setIndex(index)
+      }
+  )
```

Comment:
The [`useMemo` docs](https://reactjs.org/docs/hooks-reference.html#usememo) says:

> If no array is provided, a new value will be computed on every render.

It seems here the function will be redefined on every render, so useMemo may be useless.

Since we are more and more using hooks, I think we should add [eslint-plugin-react-hook](https://www.npmjs.com/package/eslint-plugin-react-hooks#installation) to our eslint config, so the deps of our hooks are automatically checked. See https://github.com/cozy/cozy-libs/issues/916

Reply:
Agree with the above, but I believe it's a situation for `useCallback` here?

Reply:
Ah yes, useCallback is better here as we do not have to have a `() => function () {}` but only `function () {}`. Thanks !